recaf-plugin/src/main/java/dev/xdark/recaf/plugin/RecafPluginManager.java

```diff
@@ -0,0 +1,135 @@
+	private static final String[] SUFFIX = {".jar"};
+
+	private RecafPluginManager() {
+		registerLoader(new ZipPluginLoader(RecafPluginManager.class.getClassLoader()));
+	}
+
+	@Override
+	public void unloadPlugin(String name) {
+		optionalGetPlugin(name).ifPresent(pc -> {
+			super.unloadPlugin(name);
+			pluginContainerPathMap.remove(pc);
+		});
+	}
+
+	/**
+	 * @return Root instance.
+	 */
+	public static RecafPluginManager getInstance() {
+		return Objects.requireNonNull(INSTANCE, "Please call the #initialize() first");
+	}
+
+	/**
+	 * @param path
+	 * 		The path where the plugin is located
+	 *
+	 * @return Return to PluginContainer after loading
+	 */
+	public Optional<PluginContainer<Plugin>> loadPlugin(Path path) {
+		try {
+			PluginContainer<Plugin> pc = loadPlugin(ByteSources.forPath(path));
+			pluginContainerPathMap.put(pc, path);
+			return Optional.of(pc);
+		} catch (PluginLoadException ex) {
+			logger.error("Failed to load plugin from path: {}", path, ex);
+		}
+		return Optional.empty();
+	}
+
+	/**
+	 * @param plugins
+	 * 		The plugin name set that needs to be activated
+	 */
+	public void enablePlugins(Set<String> plugins) {
+		plugins.forEach(name -> optionalGetPlugin(name)
+				.ifPresent(plugin -> plugin.getLoader().enablePlugin(plugin))
+		);
+	}
+
+	/**
+	 * @param name
+	 * 		Name of the plugin to get.
+	 *
+	 * @return Instance of the plugin container, or {@code null} if no plugin by the name exists.
+	 */
+	public Optional<PluginContainer<Plugin>> optionalGetPlugin(String name) {
+		return Optional.ofNullable(getPlugin(name));
+	}
+
+	/**
+	 * @return Map of plugins to the file paths they are loaded from.
+	 */
+	public Map<PluginContainer<? extends Plugin>, Path> getPluginContainerPathMap() {
+		return Collections.unmodifiableMap(pluginContainerPathMap);
+	}
+
+	/**
+	 * Initialize {@link RecafPluginManager} instance.
+	 */
+	public static void initialize() {
+		if (!initialized) {
+			INSTANCE = new RecafPluginManager();
+			try {
+				List<Path> paths = INSTANCE.scanPlugins();
+				for (Path path : paths)
+					INSTANCE.loadPlugin(path);
+			} catch (IOException ex) {
+				logger.error("Please check the plugin directory", ex);
+			}
+			initialized = true;
+		}
+	}
+
+	/**
+	 * Scan all plugins in the plugin directory.
+	 *
+	 * @return List of plugin paths.
+	 *
+	 * @throws IOException
+	 * 		When the plugin folder cannot be created.
+	 */
+	private List<Path> scanPlugins() throws IOException {
+		List<Path> pluginJarPaths = new ArrayList<>();
+		Path pluginsDir = Directories.getPluginDirectory();
+		// Ensure directory exists
+		if (!Files.isDirectory(pluginsDir))
+			Files.createDirectories(pluginsDir);
+		// Auto close the directory handle
+		try (DirectoryStream<Path> paths = Files.newDirectoryStream(pluginsDir)) {
+			for (Path pluginPath : paths) {
+				// Maybe we should support ".zip"
+				for (String suffix : SUFFIX) {
+					if (pluginPath.getFileName().toString().endsWith(suffix)) {
```

Comment:
This should iterate through all loaders and see if they support the byte source from this path

Reply:
That's done in `SimplePluginManager#loadPlugin(ByteSource source)`
```java
// Skip unsupported sources
if (!loader.isSupported(source))
	continue;
```

Reply:
You are right